ministryofjustice/correspondence_tool_staff
# CT-3890 solicitor address recipient page

## Description
Adds the ability for users to search for solicitor addresses and default relationship type on recipient details page.

## Self-review checklist



### Screenshots

#### Before
<img width="835" alt="Screenshot 2021-10-26 at 10 37 06" src="https://user-images.githubusercontent.com/13377553/138852257-2613410e-c162-432c-8c03-bd9a295455cd.png">

#### After

##### Initial page
<img width="814" alt="Screenshot 2021-10-26 at 10 34 11" src="https://user-images.githubusercontent.com/13377553/138851792-7be72c56-d35b-49bc-8842-ac53cf29f9a4.png">

##### Address search
<img width="804" alt="Screenshot 2021-10-26 at 10 34 47" src="https://user-images.githubusercontent.com/13377553/138851881-cbb57a44-f1c8-47f1-8862-075587a76606.png">
After selection:
<img width="847" alt="Screenshot 2021-10-26 at 10 35 17" src="https://user-images.githubusercontent.com/13377553/138851957-1f7c2ec1-c20c-46b9-911e-7131e471dae6.png">

##### Default to solicitor
<img width="867" alt="Screenshot 2021-10-26 at 10 35 56" src="https://user-images.githubusercontent.com/13377553/138852056-20284ec5-c34a-4d9b-9b0b-862ef286e465.png">

##### Select other type
<img width="873" alt="Screenshot 2021-10-26 at 10 36 25" src="https://user-images.githubusercontent.com/13377553/138852163-5d272755-ec7d-4282-96b3-2ba021338742.png">

### Related JIRA tickets
[CT-3890](https://dsdmoj.atlassian.net/browse/CT-3890)

### Manual testing instructions
- create new OffSar case
- Go to requester details step
- Select No
- Select Third Party
- Should default to Solicitor and relationship text box not present
- Search for address and select one
- When selected company name should be added from address as well
- carry on with entering case as normal
